nucypher/blockchain/eth/sol/source/contracts/lib/SignatureVerifier.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity ^0.5.3;
```

Comment:
It looks like we're ultimately passing the important stuff into `ecrecover`.  It's not immediately clear to me why this wrapper is necessary; maybe I'm just getting review fatigue.  :-)

Reply:
Reads more natural to me (e.g. `verify` returns a boolean instead of an Ethereum address which logically makes no sense). Plus, hashing and parsing signatures is embedded.
IMO would be less readable when not wrapped and when r/s/v are parsed in Python

Reply:
But I agree that it's a really, really dense read...

Reply:
As @michwill points out, `ecrecover` expects the individual `r`, `s` and `v` components the signature, the message already prehashed, and outputs an ETH address corresponding to the public key. The `SignatureVerifier` wrapper is makes things easier by allowing to pass the original message, the signature as `bytes` (which is how we'll have it most of the time) and the verifying public key.

Reply:
...but then aren't we paying gas for this small operations when we can just as easily do them in python?

Reply:
we must have signatures verification in contract and of course better to just get `r`, `s`, `v`. But if we try to put it in method `evaluateCFrag` - then will be not enough variables in stack

Reply:
Ahh, so this is more for scope hygiene than readability?

Reply:
at first place yes
but also could be only 16 local variables at once, so we can't put everything in separate variables

Reply:
Solidity (or more correctly, the EVM) sucks on this regard. The "Stack too deep" forces us to do weird and ugly things to fit everything in the stack space.